code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/lb/LoadBalancerConfigSetListenersActionHandler.java

```diff
@@ -1,42 +1,50 @@
-package io.cattle.platform.process.lb;
+package io.cattle.platform.iaas.api.lb;
 
 import static io.cattle.platform.core.model.tables.LoadBalancerConfigListenerMapTable.LOAD_BALANCER_CONFIG_LISTENER_MAP;
+import io.cattle.platform.api.action.ActionHandler;
 import io.cattle.platform.core.constants.LoadBalancerConstants;
 import io.cattle.platform.core.dao.GenericMapDao;
 import io.cattle.platform.core.model.LoadBalancerConfig;
 import io.cattle.platform.core.model.LoadBalancerConfigListenerMap;
 import io.cattle.platform.core.model.LoadBalancerListener;
-import io.cattle.platform.engine.handler.HandlerResult;
-import io.cattle.platform.engine.process.ProcessInstance;
-import io.cattle.platform.engine.process.ProcessState;
 import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
 import io.cattle.platform.object.util.DataAccessor;
-import io.cattle.platform.process.common.handler.AbstractObjectProcessHandler;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
 
 import java.util.ArrayList;
 import java.util.List;
 
 import javax.inject.Inject;
-import javax.inject.Named;
 
-@Named
-public class LoadBalancerConfigSetListeners extends AbstractObjectProcessHandler {
+public class LoadBalancerConfigSetListenersActionHandler implements ActionHandler {
 
     @Inject
     JsonMapper jsonMapper;
 
     @Inject
     GenericMapDao mapDao;
+    
+    @Inject
+    ObjectManager objectManager;
+
+    @Inject
+    ObjectProcessManager objectProcessManager;
 
     @Override
-    public String[] getProcessNames() {
-        return new String[] { LoadBalancerConstants.PROCESS_LB_CONFIG_SET_LISTENERS };
+    public String getName() {
+        return LoadBalancerConstants.PROCESS_LB_CONFIG_SET_LISTENERS;
     }
 
     @Override
-    public HandlerResult handle(ProcessState state, ProcessInstance process) {
-        LoadBalancerConfig config = (LoadBalancerConfig) state.getResource();
-        List<? extends Long> newListenerIds = DataAccessor.fromMap(state.getData()).withKey(LoadBalancerConstants.FIELD_LB_LISTENER_IDS).asList(jsonMapper,
+    public Object perform(String name, Object obj, ApiRequest request) {
```

Comment:
I dont know what happened, removing a listener was resulting in a port.create process to kick off and that was failing. But I cannot repro.